### ISSUE TITLE
MODE: Unsupported FCU with offb_node

### ISSUE THREAD

**Issue by SarahMars** [mavlink/mavros#878]
----

### Issue details
Hello,
I am trying to do offboard control with a pixracer. I want to fly it indoor so it has px4flow sensor and I am using telemetry modules(mRo SiK Telemetry Radio 915MHz). I have already tried offb_node simulation with gazebo and had no problem with it. Although I can't do it with an actual copter.

First, I opened Qgroundcontrol and I could see it connected with quadcopter through telemetry.
Second, I launched  MAVROS with
`roslaunch mavros px4.launch fcu_url:="udp://:14540@127.0.0.1:14557"`
Third, with a new terminal, I typed 
`rosrun mavros offb_node`
and then motors didn't move and nothing happened.
I just got `MODE: Unsupported FCU`
Am I doing some steps wrong?

Also when I run mavros_node,  MAVROS will shut down even I didn't change node name. (I checked offb_node and mavros_node have different name inside source code.)

```
[ WARN] [1512076026.741290904]: Shutdown request received.
[ WARN] [1512076026.741376126]: Reason given for shutdown: [new node registered with same name]
================================================================================REQUIRED process [mavros-2] has died!
process has finished cleanly
log file: /home/sarah/.ros/log/6692cc4a-d612-11e7-905c-606c66107b1f/mavros-2*.log
Initiating shutdown!
================================================================================
```

Thank you,
Sarah

### MAVROS version and platform

Mavros: 0.20.0
ROS: Lunar
Ubuntu: 16.04

### Autopilot type and version
PX4
Version: I don't know which version it is

### Diagnostics

```
header: 
  seq: 43
  stamp: 
    secs: 1512075078
    nsecs: 829763340
  frame_id: ''
status: 
  - 
    level: 1
    name: "mavros: FCU connection"
    message: "not connected"
    hardware_id: "udp://:14540@127.0.0.1:14557"


```

**Comment by vooon** [comment]
```
    name: "mavros: FCU connection"
    message: "not connected"
```

**Comment by TSC21** [comment]
@SarahMars I don't think your setup is right. You are using QGC as intermediary between MAVROS and the FCU. You should use MAVROS instead as intermediary between the FCU and QGC. So you should set the FCU url to your serial link, and the QGC URL (on the launch file) to something like `udp://:14550@localhost:14557`

**Comment by TSC21** [comment]
Also, since you are using a telemetry link, make sure to set the `SYS_COMPANION` computer to 57600, while the serial link to `/dev/ttyACM0:57600`.

**Comment by SarahMars** [comment]
@TSC21 
Thank you for your reply. I checked SYS_COMPANION on QGC and it was OSD (57600 baud 8N1).

If I change this line `<arg name="gcs_url" default=""/> ` in my px4.launch file to `<arg name="gcs_url" default="udp://:14550@127.0.0.1:14557"/> ` , and then when I run `rosrun mavros px4.launch`
I get this error.

`[ INFO] [1512149289.750564794]: serial0: device: /dev/ttyACM0 @ 57600 bps
[FATAL] [1512149289.750884993]: FCU: DeviceError:serial:open: No such file or directory
================================================================================REQUIRED process [mavros-2] has died!
process has finished cleanly
log file: /home/sarah/.ros/log/ffbd1cfe-d6bc-11e7-aae1-606c66107b1f/mavros-2*.log
Initiating shutdown!
================================================================================
[mavros-2] killing on exit
[rosout-1] killing on exit
`

Here is my px4.launch code just in case.
`<launch>
	
	

	<arg name="fcu_url" default="/dev/ttyACM0:57600" />
	<arg name="gcs_url" default="udp://:14550@127.0.0.1:14557"/> 

	<arg name="tgt_system" default="1" />
	<arg name="tgt_component" default="1" />
	<arg name="log_output" default="screen" />

	<include file="$(find mavros)/launch/node.launch">
		<arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
		<arg name="config_yaml" value="$(find mavros)/launch/px4_config.yaml" />

		<arg name="fcu_url" value="$(arg fcu_url)" />
		<arg name="gcs_url" value="$(arg gcs_url)" />
		<arg name="tgt_system" value="$(arg tgt_system)" />
		<arg name="tgt_component" value="$(arg tgt_component)" />
		<arg name="log_output" value="$(arg log_output)" />
	</include>
</launch>`

**Comment by TSC21** [comment]
> `FCU: DeviceError:serial:open: No such file or directory`

If not `ACM0`, can be `USB0`. Check on `/dev/` for `ACM0` or `USB0`.

**Comment by SarahMars** [comment]
@TSC21 
I got this after changing it to USB0. By the way where is /dev/ located at?
`[ INFO] [1512150367.312024209]: serial0: device: /dev/ttyUSB0 @ 57600 bps
[FATAL] [1512150367.312338661]: FCU: DeviceError:serial:open: Device or resource busy
REQUIRED process [mavros-2] has died!
process has finished cleanly
log file: /home/sarah/.ros/log/8203b0c2-d6bf-11e7-aae1-606c66107b1f/mavros-2*.log
Initiating shutdown!
`

**Comment by TSC21** [comment]
You cannot have QGC to automatically connect to the FCU. Go to QGC settings and disable the option to connect to the `Pixhawk` and the `SiK Radio`.

**Comment by TSC21** [comment]
![qgc general settings](https://user-images.githubusercontent.com/5048656/33495927-2c641d72-d6c0-11e7-91c6-88389db17ee8.png)

**Comment by TSC21** [comment]
`By the way where is /dev/ located at?`

On your system's root.

**Comment by SarahMars** [comment]
@TSC21 
After disabling pixhawk and SiK radio, I got this

```[ INFO] [1512150802.265587539]: serial0: device: /dev/ttyUSB0 @ 57600 bps
[ INFO] [1512150802.266750092]: GCS URL: udp://:14550@127.0.0.1:14557
[ERROR] [1512150802.266814919]: serial0: receive: End of file
terminate called after throwing an instance of 'std::system_error[ INFO] [1512150802.267336995]: udp1: Bind address: 0.0.0.0:14550
'
  what():  Resource deadlock avoided
================================================================================REQUIRED process [mavros-2] has died!
process has died [pid 11190, exit code -6, cmd /home/sarah/catkin_ws5/devel_isolated/mavros/lib/mavros/mavros_node __name:=mavros __log:=/home/sarah/.ros/log/8541fe1e-d6c0-11e7-aae1-606c66107b1f/mavros-2.log].
log file: /home/sarah/.ros/log/8541fe1e-d6c0-11e7-aae1-606c66107b1f/mavros-2*.log
Initiating shutdown!
```

**Comment by TSC21** [comment]
https://github.com/mavlink/mavros/blob/master/mavros/README.md#troubleshooting

**Comment by SarahMars** [comment]
Sorry I'm still searching for how to check /dev/ (I'm new to linux)

**Comment by TSC21** [comment]
> Sorry I'm still searching for how to check /dev/ (I'm new to linux)

`cd /dev`. Lots of docs online. Start with for example: https://www.ibm.com/developerworks/library/l-lpic1-103-1/index.html

**Comment by SarahMars** [comment]
Thank you for the example website, I'll take a look at it.

I've tried https://github.com/mavlink/mavros/blob/master/mavros/README.md#troubleshooting , now it gave me
```[ INFO] [1512151658.944483415]: serial0: device: /dev/ttyUSB0 @ 57600 bps
[FATAL] [1512151658.944907999]: FCU: DeviceError:serial:open: No such file or directory
================================================================================REQUIRED process [mavros-2] has died!
process has finished cleanly
log file: /home/sarah/.ros/log/83e340e4-d6c2-11e7-aae1-606c66107b1f/mavros-2*.log
Initiating shutdown!
================================================================================```

cd/dev/ls is like this

![image](https://user-images.githubusercontent.com/32495711/33496732-5de473dc-d699-11e7-94fb-654bdb1f4494.png)

**Comment by TSC21** [comment]
You have `ACM0` available.

**Comment by SarahMars** [comment]
Oh I `ttyACM0`, I got it.

Thank you so much for your help, I can now connect to the FCU!
``` 
    level: 0
    name: "mavros: FCU connection"
    message: "connected"
    hardware_id: "/dev/ttyACM0:57600"
```

**Comment by TSC21** [comment]
OK. QGC got the connection also?

**Comment by SarahMars** [comment]
Oh wait, I wasn't opening QGC while running the command.

```
[ INFO] [1512153362.161605754]: serial0: device: /dev/ttyACM0 @ 57600 bps
[ INFO] [1512153362.162249010]: GCS URL: udp://:14550@127.0.0.1:14557
[ INFO] [1512153362.162880156]: udp1: Bind address: 0.0.0.0:14550
[ INFO] [1512153362.163025246]: udp1: Remote address: 127.0.0.1:14557
[FATAL] [1512153362.163568876]: GCS: DeviceError:udp:bind: Address already in use
```

**Comment by TSC21** [comment]
try `udp-b://127.0.0.1:14555@14550`

**Comment by SarahMars** [comment]
It worked! Thank you so much!

Although I could get connection when the USB cable is connected, how am I supposed to get connection when I want to fly it?

**Comment by TSC21** [comment]
> Although I could get connection when the USB cable is connected, how am I supposed to get connection when I want to fly it?

What do you mean? Don't you have a telemetry module?

**Comment by SarahMars** [comment]
@TSC21 
I was connecting both of telemetry module and USB cable. I connected both of them because it didn't connect without a USB cable. Sorry now I understood I was doing weird thing.

If I only plug in telemetry module, I get `Error: serial0: receive: End of file`. Even if I followed the steps given in [https://github.com/mavlink/mavros/blob/master/mavros/README.md#troubleshooting](url), it will still give me this error.

Also if I only connect telemetry module, it was `ttyUSB0` so I changed px4.launch file.
